src/daemon/distribution_test.c

```diff
@@ -0,0 +1,812 @@
+/**
+ * collectd - src/daemon/distribution_test.c
+ * Copyright (C) 2020       Google LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ * Author:
+ *   Elene Margalitadze <elene.margalit at gmail.com>
+ */
+
+#include "../testing.h"
+#include "collectd.h"
+#include "distribution.c"
+
+static double *initialize_linear_min_bounds(size_t num_buckets, double size) {
+  double *arr = malloc(num_buckets * sizeof(double));
+  for (size_t i = 0; i < num_buckets; i++) {
```

Comment:
I'd recommend to check if arr is not equal to NULL because if malloc could fail and if yes, then the program would crash. The reason is that the program would try to read some data from null pointer.

That applies also to the other functions for initialising bounds.

Reply:
Right, thank you Barbara for pointing that out!